compiler/rustc_middle/src/query/mod.rs

```diff
@@ -721,6 +721,11 @@ rustc_queries! {
         desc { |tcx| "computing drop-check constraints for `{}`", tcx.def_path_str(key) }
     }
 
+    query movability(key: DefId) -> hir::Movability {
+        desc { |tcx| "checking if coroutine is movable: `{}`", tcx.def_path_str(key) }
+        separate_provide_extern
+    }
```

Comment:
Does this need to be a query, or can it be computed from coroutine_kind ?

Reply:
It doesn't need to be after #119258 -- then it can be replaced with `tcx.coroutine_kind(def_id).expect("...").movability()` -- but that PR has not landed yet.

Reply:
I can make that resulting change to either this PR or #119258, but for now I'm waiting for either of those to be r+'d.

Reply:
#119258 has been merged

Reply:
I've implemented this change in 3a9ec2109bd55bd4ffc0db727efcb730a8e19bae, though I wonder if we should instead keep `tcx.movability` around as a convenience function... maybe rename it to `tcx.closure_movability`?

Reply:
Yes. A coroutine_movability helper sounds useful.